Services/CO.CDP.Organisation.WebApi/UseCase/UpdateOrganisationUseCase.cs

```diff
@@ -188,12 +195,70 @@ public async Task<bool> Execute((Guid organisationId, UpdateOrganisation updateO
 
         organisation.UpdateSupplierInformation();
 
+        await ResetRejectedStatus(command.updateOrganisation.Type, organisation);
+
         await organisationRepository.SaveAsync(organisation,
             async o => await publisher.Publish(mapper.Map<OrganisationUpdated>(o)));
 
         return await Task.FromResult(true);
     }
 
+    private async Task ResetRejectedStatus(OrganisationUpdateType updateType, Persistence.Organisation organisation)
+    {
+        if (
+            (updateType == OrganisationUpdateType.OrganisationName || updateType == OrganisationUpdateType.OrganisationEmail)
+            && organisation.PendingRoles.Contains(PartyRole.Buyer)
+            && organisation.ReviewedBy != null
+        )
+        {
+            await ResendBuyerApprovalEmail(organisation);
+
+            organisation.ReviewComment = null;
+            organisation.ReviewedBy = null;
+        }
+    }
+
+    private async Task ResendBuyerApprovalEmail(Persistence.Organisation organisation)
+    {
+        var baseAppUrl = configuration.GetValue<string>("OrganisationAppUrl") ?? "";
```

Comment:
Would be nice, if we have all EmailNotification related logic (templateId, reading configs and building EmailNotificationRequest object) encapsulated away into a separate EmailService class, 
where all we need to call the method and required parameter like below - 

`EmailService.SendBuyerApprovalEmail(string organisationName, string requestLink);`

Reply:
I understand this change may not lie under current story scope, but we can start moving this method and later other email notification methods one by one.

Reply:
yeah the amount of duplication has got a bit out of hand there hasn't it. I'd be happy to do it, but yeah perhaps not in this change right now.

Next time we modify the email code, definitely...